internal/controller/controllers/agent_controller.go

```diff
@@ -317,6 +319,19 @@ func (r *AgentReconciler) updateStatus(ctx context.Context, log logrus.FieldLogg
 				return ctrl.Result{Requeue: true}, nil
 			}
 		}
+		if agent.Status.DebugInfo.LogsURL == "" {
+			logsCollected := areLogsCollected(h.LogsCollectedAt)
+			if logsCollected {
```

Comment:
This can all be one line:
`if agent.Status.DebugInfo.LogsURL == "" && areLogsCollected(h.LogsCollectedAt)`

Even with @flaper87 's suggestion (removing the `areLogsCollected` function), I think this would be fine as one line and a comment.

Reply:
Good idea, I've modified the code to reflect this. Thanks!